### PR TITLE
Added Login component for PPR and other apps.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #:*
https://github.com/bcgov/entity/issues/10248

*Description of changes:*
1. Added login component to display generic login page for PPR and other apps;
2. Fixed layout.scss got overwrite issue;

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: vue/sbc-common-components/src/components/SbcAuthMenu.vue [hunk 1]
```diff
@@ -154,13 +155,65 @@ export default class SbcAuthMenu extends Mixins(NavigationMixin) {
     }
   }
 
-  login (idpHint) {
-    if (this.redirectOnLoginSuccess) {
-      let url = encodeURIComponent(this.redirectOnLoginSuccess)
-      url += this.redirectOnLoginFail ? `/${encodeURIComponent(this.redirectOnLoginFail)}` : ''
-      window.location.assign(`${this.getContextPath()}signin/${idpHint}/${url}`)
+  // login (idpHint) {
```
Comment (severinbeauvais) [inline vue/sbc-common-components/src/components/SbcAuthMenu.vue:158]: Delete this.

File: vue/sbc-common-components/src/components/SbcLogin.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                  large
+                  class="white--text font-weight-bold"
+                  aria-label="log in"
+                  id="loginBtn"
+                  v-on="on">
+                  <span>Log in to my BC Registries Account</span>
+                  <v-icon class="mr-n1 ml-2">mdi-menu-down</v-icon>
+                </v-btn>
+              </template>
+              <sbc-auth-menu :fromLogin="true" :redirect-on-login-success="redirectUrl" />
```
Comment (severinbeauvais) [inline vue/sbc-common-components/src/components/SbcLogin.vue:32]: Does the signin button on the BCROS pages need to specify any of these props when it uses the SbcAuthMenu component?

image

File: vue/sbc-common-components/src/components/SbcAuthMenu.vue [hunk 3]
```diff
@@ -30,12 +30,12 @@ import { LDClient } from 'launchdarkly-js-client-sdk'
 import { IdpHint, LoginSource, Pages } from '../util/constants'
```
Comment (severinbeauvais) [inline vue/sbc-common-components/src/components/SbcAuthMenu.vue:30]: Do the BCROS pages (my PR) need to use the new SbcAuthMenu component?

File: vue/sbc-common-components/src/components/SbcLogin.vue (hunk 2 above)
Comment (severinbeauvais) [reply vue/sbc-common-components/src/components/SbcLogin.vue:32]: I mean, do I need to specify those props from BCROS pages?

File: vue/sbc-common-components/src/components/SbcAuthMenu.vue (hunk 3 above)
Comment (severinbeauvais) [reply vue/sbc-common-components/src/components/SbcAuthMenu.vue:30]: Patrick responded:
> You need to test it. I just added an option to it. The original way doesn’t work for the PPR application.

Comment (severinbeauvais) [reply vue/sbc-common-components/src/components/SbcAuthMenu.vue:30]: The BCROS pages appear to work correctly without the new SbcAuthMenu component.

File: vue/sbc-common-components/src/components/SbcLogin.vue (hunk 2 above)
Comment (pwei1018) [reply vue/sbc-common-components/src/components/SbcLogin.vue:32]: You may need to test the bookmark logic. If user bookmark a specify page that need to login (i.e. dashboard), your BCROS page should handle it.

File: vue/sbc-common-components/src/components/SbcAuthMenu.vue (hunk 1 above)
Comment (pwei1018) [reply vue/sbc-common-components/src/components/SbcAuthMenu.vue]: Done.

File: vue/sbc-common-components/src/components/SbcLogin.vue (hunk 2 above)
Comment (severinbeauvais) [reply vue/sbc-common-components/src/components/SbcLogin.vue:32]: Yeah, that's working. Thanks for the feedback.

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**